File: Artist/app/src/test/java/com/example/artist/TopAlbumsPresenterTest.java
package com.example.artist;

import com.example.artist.ui.topalbumslisting.TopAlbumsPresenter;
import com.example.artist.ui.topalbumslisting.TopAlbumsPresenterImpl;
import com.example.artist.utils.Constants;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class TopAlbumsPresenterTest {
    private TopAlbumsPresenter topAlbumsPresenter = null;



    @Before
    public void setUp() {
        topAlbumsPresenter = new TopAlbumsPresenter() {
            @Override
            public void onDestroy() {

            }

            @Override
            public void getTopAlbums(String userName, String apiKey) {

            }
        };

    }

    @After
    public void tearDown() {

    }

    @Test
    public void testGetTopAlbums() {
        topAlbumsPresenter.getTopAlbums("b", Constants.API_KEY);

    }
}

File: Artist/app/src/test/java/com/example/artist/TopAlbumsPresenterTest.kt
package com.example.artist

import com.example.artist.ui.topalbumslisting.TopAlbumsFragment
import com.example.artist.ui.topalbumslisting.TopAlbumsInteractorImpl
import com.example.artist.ui.topalbumslisting.TopAlbumsPresenterImpl
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito

class TopAlbumsPresenterTest {
    lateinit var topAlbumsPresenter: TopAlbumsPresenterImpl

    @Before
    fun setUp() {
        val fragmentMock = Mockito.mock(TopAlbumsFragment::class.java)
        val topAlbumsInteractorMock = Mockito.mock(TopAlbumsInteractorImpl::class.java)
        topAlbumsPresenter = TopAlbumsPresenterImpl(fragmentMock, topAlbumsInteractorMock)
    }

    @After
    fun tearDown() {
    }

    @Test
    fun test_getTopAlbums_calls_the_update_method_once_on_view() {
        Mockito.`when`(topAlbumsPresenter.getTopAlbums("b", "mockApiKey")).then {
            Mockito.verify(topAlbumsPresenter.mView.updateData(mutableListOf()), Mockito.times(1))
        }
    }
}